SimTKcommon/src/ParallelExecutor.cpp

```diff
@@ -25,23 +25,36 @@
 #include "SimTKcommon/internal/ParallelExecutor.h"
 #include <pthread.h>
 
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+using namespace std;
+
 namespace SimTK {
 
 static void* threadBody(void* args);
 
+ParallelExecutorImpl::ParallelExecutorImpl() : finished(false) {
+
+    // Set the maximum number of threads that we can use
+    SimTK_APIARGCHECK_ALWAYS(ParallelExecutor::getNumProcessors() > 0, "ParallelExecutorImpl", "ParallelExecutorImpl", "Number of threads detected is not positive.");
```

Comment:
That error check macro is for API arguments. This constructor doesn't have any arguments. If getNumProcessors() is querying the system and it comes back zero, that's a disaster, not a bad argument!